SSA/Core/Framework.lean

```diff
@@ -2589,6 +2589,65 @@ theorem denote_rewritePeephole (fuel : ℕ)
 /-- info: 'denote_rewritePeephole' depends on axioms: [propext, Classical.choice, Quot.sound] -/
 #guard_msgs in #print axioms denote_rewritePeephole
 
+variable {d : Dialect} [DialectSignature d] [DecidableEq (Dialect.Ty d)] [DecidableEq (Dialect.Op d)]
+[TyDenote d.Ty] [DialectDenote d] [Monad d.m] in
+/--  rewrite with the list of peephole optimizations `prs` at the `target` program, at location `ix`
+and later, running at most `fuel` steps. -/
+def rewritePeephole_go_multi (fuel : ℕ) (prs : List (PeepholeRewrite d Γ t))
+    (ix : ℕ) (target : Com d Γ₂ eff t₂) : Com d Γ₂ eff t₂ :=
+  match fuel with
+  | 0 => target
+  | fuel' + 1 =>
+    let target' := prs.foldl (fun acc pr => rewritePeepholeAt pr ix acc) target
+    rewritePeephole_go_multi fuel' prs (ix + 1) target'
+
+variable {d : Dialect} [DialectSignature d] [DecidableEq (Dialect.Ty d)] [DecidableEq (Dialect.Op d)]
+[TyDenote d.Ty] [DialectDenote d] [Monad d.m] in
+/-- rewrite with the list of peephole optimizations `prs` at the `target` program, running at most
+`fuel` steps starting at location 0. -/
+def rewritePeephole_multi (fuel : ℕ)
```

Comment:
```suggestion
def multiRewritePeepholeAt (fuel : ℕ)
```

Naming convention for such definition is to use lowerCamelCase. Since you use `rewritePeepholeAt` it's better to reflect that fully in the name. Finally, since `rewritePeepholeAtMulti` could suggest multiple locations, I suggest moving the multi to the front so it's a bit less ambiguous.